### ISSUE TITLE
BUG: Can not run 'AttributeError: module 'numpy.__config__' has no attribute 'get_info''

### ISSUE THREAD

**Issue by Gabriel-p** [pymc-devs/pymc#7043]
### Describe the issue:

Followed the instructions to install:

```
$ conda create -c conda-forge -n pymc_env "pymc>=5"
```

but can not even import the package.

### Reproduceable code example:

```python
import pymc
```

### Error message:

```shell
Traceback (most recent call last):
  File "/home/gabriel/miniconda3/envs/pymc_env/lib/python3.11/site-packages/pytensor/configparser.py", line 201, in fetch_val_for_key
    return self._pytensor_cfg.get(section, option)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gabriel/miniconda3/envs/pymc_env/lib/python3.11/configparser.py", line 797, in get
    d = self._unify_values(section, vars)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gabriel/miniconda3/envs/pymc_env/lib/python3.11/configparser.py", line 1168, in _unify_values
    raise NoSectionError(section) from None
configparser.NoSectionError: No section: 'blas'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/gabriel/miniconda3/envs/pymc_env/lib/python3.11/site-packages/pytensor/configparser.py", line 317, in __get__
    val_str = cls.fetch_val_for_key(self.name, delete_key=delete_key)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gabriel/miniconda3/envs/pymc_env/lib/python3.11/site-packages/pytensor/configparser.py", line 205, in fetch_val_for_key
    raise KeyError(key)
KeyError: 'blas__ldflags'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/gabriel/miniconda3/envs/pymc_env/lib/python3.11/site-packages/pymc/__init__.py", line 47, in <module>
    __set_compiler_flags()
  File "/home/gabriel/miniconda3/envs/pymc_env/lib/python3.11/site-packages/pymc/__init__.py", line 30, in __set_compiler_flags
    import pytensor
  File "/home/gabriel/miniconda3/envs/pymc_env/lib/python3.11/site-packages/pytensor/__init__.py", line 119, in <module>
    from pytensor import scalar, tensor
  File "/home/gabriel/miniconda3/envs/pymc_env/lib/python3.11/site-packages/pytensor/tensor/__init__.py", line 106, in <module>
    from pytensor.tensor import (  # noqa
  File "/home/gabriel/miniconda3/envs/pymc_env/lib/python3.11/site-packages/pytensor/tensor/blas.py", line 102, in <module>
    from pytensor.tensor.blas_headers import blas_header_text, blas_header_version
  File "/home/gabriel/miniconda3/envs/pymc_env/lib/python3.11/site-packages/pytensor/tensor/blas_headers.py", line 1015, in <module>
    if not config.blas__ldflags:
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/gabriel/miniconda3/envs/pymc_env/lib/python3.11/site-packages/pytensor/configparser.py", line 321, in __get__
    val_str = self.default()
              ^^^^^^^^^^^^^^
  File "/home/gabriel/miniconda3/envs/pymc_env/lib/python3.11/site-packages/pytensor/link/c/cmodule.py", line 2720, in default_blas_ldflags
    blas_info = np.__config__.get_info("blas_opt")
                ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'numpy.__config__' has no attribute 'get_info'
```

### PyMC version information:

```
  pymc               pkgs/main/linux-64::pymc-5.6.1-py311ha39b09d_0 
  pytensor           pkgs/main/linux-64::pytensor-2.12.3-py311ha02d727_0 
  python             conda-forge/linux-64::python-3.11.6-hab00c5b_0_cpython 
```

Used conda to install.

**Comment by EliasRas** [comment]
Are you using numpy==1.26.0? Downgrading to numpy 1.25.2 worked for me. The newest numpy version (https://numpy.org/doc/stable/release/1.26.0-notes.html) has a different build system which might be the reason.

**Comment by Gabriel-p** [comment]
I'm using version 1.26.2 which is the version `pymc` requested apparently. Downgrading to 1.25.2 apparently fixed the issue, thank you.

Now the new issue is: why would pymc request an incompatible numpy version?

**Comment by ricardoV94** [comment]
PyTensor doesn't have an upper pin on numpy (most libraries don't ) so it just installed the latest. That version however is not compatible.

If you install the latest pymc, the respective pytensor is already compatible with the latest numpy

**Comment by ricardoV94** [comment]
@maresb is there something we can do about the defaults channel? Why does it exist and to an older version?

**Comment by maresb** [comment]
Defaults is Anaconda's channel, which is generally slower than conda-forge.

I'd recommend adding `nodefaults` to the channels list to prevent mixing.

**Comment by joeyamosjohns** [comment]
@EliasRas Would you be so kind as to post how to install specific version of numpy?
I tried this but pymc still gave same error:

```
import pkg_resources
pkg_resources.require("numpy==1.25.2")  # modified to use specific numpy
import numpy
```
@ricardoV94 I am happy to install pymc 5.10 ... but I am a beginner and following the current install instructions here: https://www.pymc.io/projects/docs/en/stable/installation.html#installation
Can you please post similar install instructions here? (like do I just change "pymc>=5" to "pymc 5.10" in this:

```
conda create -c conda-forge -n pymc_env "pymc>=5"
conda activate pymc_env
```

**Comment by maresb** [comment]
Hi @joeyamosjohns, you can try

```bash
conda create -c conda-forge -n pymc_env "pymc>=5.10" "numpy=1.25.2"
conda activate pymc_env
```

**Comment by maresb** [comment]
It would help if you provided some more information. Like which operating system are you using?

A few more things you could try:

```
conda create -c conda-forge -c nodefaults -n pymc_env "pymc>=5.10" "numpy=1.25.2"
conda activate pymc_env
```

If that still doesn't work, then please provide us with the output of the following commands after activating:

```
conda config --show
conda list
```

**Comment by joeyamosjohns** [comment]
Thank you kindly @maresb ... I will get back to you this evening

**Comment by joeyamosjohns** [comment]
@maresb 
Ben, I was delayed until weekend, but your last suggestion worked! (For the record, I changed"pymc>=5.10" to  "pymc=5.10" when I tried it, because >= sounds like it might allow some version higher than 5.10). Thanks a million, Ben, I might not have got it installed without you (and missed my destiny of becoming a Grand Master Bayesian haha). Cheers.

**Comment by maresb** [comment]
Thanks so much for the update @joeyamosjohns!

The fact that `-c nodefaults` helped seems to indicate that you have the `defaults` channel configured in such a way that it's taking priority over the `conda-forge` channel. To ensure that you don't have problems going forward you may want to check out [these instructions from conda-forge](https://conda-forge.org/docs/user/tipsandtricks.html#how-to-fix-it).

All the best on your quest!!!

**Comment by Sheref001** [comment]
Thank you so much @maresb for your recommendation! I wanted to mention that I encountered some problems while attempting to install Jupyter in the pymc_env. The error message stated: "package llvm-tools-16.0.6-hbedff68_3 requires libxml2 >=2.12.1,<3.0.0a0, but none of the providers can be installed." However, I was able to resolve this issue by running the command "conda config --add channels conda-forge" and then installing Jupyter. It appears that everything is functioning properly now.
